### PR TITLE
Fixes some error with the services

### PR DESCRIPTION
**Pull request by GitHK**
### Fix failure while recovering ports information
While recovering node port information from the database, the case in which the is missing was never handled. The PR addresses the issue.
Messages on failures will also be sent to the UI for easier debugging in the future.

### Fix uncaught exception
`DatabaseError`was not caught and caused an exception if the pipeline was missing.

### Fix uncaught exception in sidecar pipeline
`DockerError` was never caught. When pulling an image (if the docker registry is not responding) an exception is raised and never handled.

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: packages/simcore-sdk/src/simcore_sdk/node_ports/exceptions.py [hunk 1]
```diff
@@ -1,95 +1,138 @@
...
+
     def __init__(self, msg):
         super(PortNotFound, self).__init__(msg)
+
+
+class NodeNotFond(NodeportsException):
+    """Accessed key does not exist"""
+
+    def __init__(self, msg):
+        super(NodeNotFond, self).__init__(msg)
```
Comment (sanderegg) [inline packages/simcore-sdk/src/simcore_sdk/node_ports/exceptions.py:138]: maybe pass de node uuid in there

File: packages/simcore-sdk/src/simcore_sdk/node_ports/dbmanager.py [hunk 2]
```diff
@@ -40,7 +41,9 @@ async def _get_node_from_db(
         log.error("the node id %s is not unique", node_uuid)
     node = await result.fetchone()
     if not node:
-        log.error("the node id %s was not found", node_uuid)
+        message = "the node id %s was not found", node_uuid
+        log.error(message)
+        raise NodeNotFond(message)
```
Comment (sanderegg) [inline packages/simcore-sdk/src/simcore_sdk/node_ports/dbmanager.py:46]: pass the node uuid in the exception

File: packages/simcore-sdk/src/simcore_sdk/node_ports/exceptions.py [hunk 3]
```diff
@@ -1,95 +1,138 @@
...
 
+
 class PortNotFound(NodeportsException):
     """Accessed key does not exist"""
+
     def __init__(self, msg):
         super(PortNotFound, self).__init__(msg)
+
+
+class NodeNotFond(NodeportsException):
```
Comment (sanderegg) [inline packages/simcore-sdk/src/simcore_sdk/node_ports/exceptions.py:134]:
```suggestion
class NodeNotFound(NodeportsException):

```

File: services/sidecar/src/simcore_service_sidecar/executor.py [hunk 4]
```diff
@@ -370,13 +377,20 @@ async def _process_task_output(self):
                 # WARNING: nodeports is NOT concurrent-safe, dont' use gather here
                 for coro in file_upload_tasks:
                     await coro
-
+        except node_ports.exceptions.NodeNotFond:
+            await self._error_message_to_ui_and_logs(
+                "Error: no ports info found in the database."
```
Comment (sanderegg) [inline services/sidecar/src/simcore_service_sidecar/executor.py:382]: use the exception message with the node uuid

**Comment by GitHK** [comment]
> Also you check here only in the process_output part. but the same could happen when processing the inputs right?

I checked for inputs and outputs. In both situations (all the places where this function is used).

**Comment by GitHK** [comment]
> 1 user gets bored and disconnects, at some point the project gets deleted. then a sidecar kicks in, and the node has disappeared. would it be possible?

Yes, because all the sidecar queues were full

**Pull request merged**